Week1 [STEP 3] Kobe

devKobe24 opened the pull request:
@Gundy93
 건디의 응원 덕분에 드디어 스탭 3에 왔습니다!!
 이 기세를 몰아 PR 보냅니다 :)

 ## 고민했던 점
 ### 같은 로직이라도 다르게 쓰는 법
 - 같은 로직이라도 굳이 똑같은 방법을 사용해서 코드를 짜지 않고 또 다른 방법은 없나 생각하며 다르게 코드를 써봤습니다.

 ---

 ## 해결이 되지 않은 점
 ### 마지막 저만의 빼빼로 만들기 함수에서 막대를 가운데로 오게 만드는 것이 힘들었습니다.
 - 토핑이 있고 없고 차이에 따라 막대가 가운데가 될 때도 있고 아닌때도 있는 거 같은데 이게 올바르게 잘 짜여진 코드인지 모르겠습니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 개발 공부 방법
 - 건디는 어떤 방법으로 개발 공부를 하시는지 궁굼합니다. 저는 주로 개발 블로그를 찾아다니며 보고 따라 쓰는 그런 방법과 책을 사서 읽고 제 블로그에 정리하는 방법으로 공부하는데 이게 잘 남지 않는 것 같습니다.

Commit:
Week1 [STEP 3] Kobe

 @Gundy93
 건디의 응원 덕분에 드디어 스탭 3에 왔습니다!!
 이 기세를 몰아 PR 보냅니다 :)

 ## 고민했던 점
 ### 같은 로직이라도 다르게 쓰는 법
 - 같은 로직이라도 굳이 똑같은 방법을 사용해서 코드를 짜지 않고 또 다른 방법은 없나 생각하며 다르게 코드를 써봤습니다.

 ---

 ## 해결이 되지 않은 점
 ### 마지막 저만의 빼빼로 만들기 함수에서 막대를 가운데로 오게 만드는 것이 힘들었습니다.
 - 토핑이 있고 없고 차이에 따라 막대가 가운데가 될 때도 있고 아닌때도 있는 거 같은데 이게 올바르게 잘 짜여진 코드인지 모르겠습니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 개발 공부 방법
 - 건디는 어떤 방법으로 개발 공부를 하시는지 궁굼합니다. 저는 주로 개발 블로그를 찾아다니며 보고 따라 쓰는 그런 방법과
   책을 사서 읽고 제 블로그에 정리하는 방법으로 공부하는데 이게 잘 남지 않는 것 같습니다.

Gundy93 reviewed (commented):
코비님! 이번 스텝은 평소보다도 파이팅 넘치게 하신 것 같습니다 👍

> 같은 로직이라도 다르게 쓰는 법

다양한 방법으로 코드를 사용할 수 있으면 좋습니다. 같은 결과를 낼 수 있는 코드의 종류는 무궁무진하니까요!

다양한 방법을 익히고 상황에 가장 적절한 로직을 선택해서 사용할 수 있는 것이 좋겠습니다

코비님은 이번 스텝에서 어떤 로직이 제일 적절한 것 같았나요?

수정시에 그 로직만 남겨주시면 더 상세하게 코멘트 드리기 좋을 것 같습니다

> 마지막 저만의 빼빼로 만들기 함수에서 막대를 가운데로 오게 만드는 것이 힘들었습니다.

아래의 개별 코멘트를 통해 해결이 되면 좋겠네요! 그리고 아이스크림과 달리 빼빼로는 막대가 가운데라기보다는 몸통의 두께와 같습니다.

![image](https://user-images.githubusercontent.com/106914201/222942596-0e0f4bef-8712-4246-9414-1866b03fac76.png)

다만 토핑이 있다면 끝선과 떨어지게 되겠죠! 그러니까 토핑과 몸통의 경계에 `"|"`을 위치시키는 것도 좋겠습니다.

> 개발 공부 방법

iOS의 경우 애플이 제공하는 방대한 분량의 아티클이나 WWDC 영상, HIG 문서등이 있는데, 해당 부분을 위주로 공부합니다. 그것만 잘 보아도 훌륭한 공부가 됩니다. 그 밖에 애플에서 따로 제공하지 않는 디자인패턴이나 아키텍쳐 등은 구글링을 통해 공부하기도 합니다. 😃

제가 코멘트를 첫 번째 로직에만 남겨두었는데, 이 코멘트는 대체로 다른 로직들에도 적용되는 코멘트입니다.

특히 `String?`을 받는 토핑의 경우 `nil`과 `non-nil`에서 모두 정상적으로 작동하는지 확인하는 것이 필요하겠습니다 🤔

이번 스텝 수정도 화이팅입니다 🙌

Commit:
Week1 [STEP 3] Kobe

 @Gundy93
 건디 수정사안 모두 반영하여 다시 PR 보냅니다 :)

 ## 고민했던 점
 ### 분기처리
 - nil 이 들어옴에 따라 빼빼로의 모양이 달라지는데 그것에 따른 분기 처리를 어떻게 할지 고민해봤습니다.

 ---

 ## 해결이 되지 않은 점
 ### 이번에는 질문-답변 채널에서 답을 얻었습니다.
 - main.swift 파일에서 직접 STEP3.swift 함수를 호출해야 한다는 점을 알게 되었습니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 조바심이 나는 부분
 - 아무래도 현업과 병행을 하다보니 다른 캠퍼분들과 달리 진도가 매우 느린것 같아 조바심이 납니다.
 다른 캠퍼분들은 이미 3주차를 진행하는데 저는 1주차도 못끝낸것 같아 매우 조바심이 납니다. ㅠㅠ

Gundy93 reviewed (commented):
코비님 이번 코멘트도 마찬가지로 모든 경우에 통용되는 코멘트니 참고해주세요

현재 플로우 차트에서 `두 번째 for문`과 같이 우측에 설명을 만드신 부분을 제외하고 만드시면 어떨까요?

해당 설명이 없더라도 흐름을 잘 이해할 수 있는 순서도를 만드는 것이 좋겠습니다.

<img width="233" alt="스크린샷 2023-03-07 오전 11 56 54" src="https://user-images.githubusercontent.com/106914201/223308617-d274f516-2386-4ae7-b33c-c15a4dd74fd3.png">

`_`라는 표현이 이 순서도에서 반복적으로 나타나는데, 아무 설명없이 직관적으로 이해하기 어려울 것 같습니다.

<img width="213" alt="스크린샷 2023-03-07 오전 11 59 11" src="https://user-images.githubusercontent.com/106914201/223308943-9a85d25a-8381-4d6e-b1a8-92b93a41a4bc.png">

`_`가 1일 때 `_`는 9보다 작으므로 아래로 내려갈텐데, 밑의 과정을 수행한 후에 다시 올라가지 않게 됩니다.

그렇기 때문에 현재는 `_가 9보다 작으면 아래의 과정을 1회만 수행`으로 이해할 것 같습니다.

<img width="131" alt="스크린샷 2023-03-07 오전 11 54 32" src="https://user-images.githubusercontent.com/106914201/223308268-e6685dc0-fbff-43fb-bbe8-b8eaf05a846c.png">

또한 이렇게 같은 포인트로 들어오는 선과 나가는 선이 겹치게 되는 것은 피하면서 만들어보면 어떨까요?

질문칸을 아래로 내리고 로직을 위로 올리면 깔끔한 순서도를 만들기 쉽습니다.

![제목 없는 다이어그램 drawio-8](https://user-images.githubusercontent.com/106914201/223308506-a8f766f6-14ef-4f2c-b0bd-b3902db2e915.png)

이번에 설명을 위해 제가 만들어본 순서도입니다. 도움이 되었으면 좋겠네요!

Commit:
Week 1 [STEP 3] Kobe

 @Gundy93
 건디 피드백 반영하여 수정했습니다 :)

 ## 고민했던 점
 ### 모든 함수를 한 번에 불러오기
 - 모든 함수를 한 번에 불러올 때 제대로 나오기는 하는데 '몸통(String), 토핑(String)을 매개변수로 받아 토핑과 몸통을 그리는 함수' 때문에 길이가 1개가 더 늘어나는 것 같습니다.

 ---

 ## 해결이 되지 않은 점
 ### 위 고민했던 점과 동일합니다.
 - 함수를 불러 올 때 분기 처리를 통하여 '몸통(String), 토핑(String)을 매개변수로 받아 토핑과 몸통을 그리는 함수'를 조건에 따라 호출되지 않도록 해야하는지 궁굼합니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 커리어 스타터 캠프에 대한 부분
 - 이번에 퇴사를 결정하고 커리어 스타터 캠프를 듣기로 결정했습니다.
 퇴사 일자는 아직 3주 정도 남았지만 그래도 매우 조급합니다 그 동안 커리어 캠프 테스트 준비를 해야하는데 못하고 있는것 같아서 말이죠...
 건디는 처음 준비를 어떻게 하셨는지 궁굼합니다.

Gundy93 reviewed (commented):
코비님! 직장을 다니며 캠프를 진행하는 것이 쉽지 않음을 압니다! 끝까지 포기하지말고 열심히 해봅시다!

> 모든 함수를 한 번에 불러오기 & 위 고민했던 점과 동일합니다.

해당 부분은 아래의 개별 코멘트를 참고해주세요!

> 커리어 스타터 캠프에 대한 부분

코드 스타터 캠프에 있는 학습자료를 잘 익힌다면 큰 무리없이 사전과제를 진행할 수 있고, 사전과제를 수행할 수 있는 기간이 생각보다 짧지 않기 때문에 큰 부담을 갖지 않으셔도 괜찮을 것 같습니다.

다만 조급한 마음에 진도를 많이 나가고자 훑듯이 본다면 제대로 지식을 습득하기 어렵기 때문에 차분히, 꼼꼼히 보시는 것이 좋겠습니다.

퇴사를 하고나면 여유가 생기실테니 차분히 캠프내용을 처음부터 따라가면서 준비하면 될 것 같습니다

이번 스텝의 수정부터 천천히 해볼까요?

이번에도 마찬가지로 첫 번째 로직에 대한 피드백을 드렸습니다. 그것을 제대로 만들 수 있다면 다른 함수에 대한 수정도 같이 하실 수 있을 것입니다.

수정하시면서 첫 번째 방법으로 모든 예시를 출력하게끔 해보시면 어딘가 잘못된 부분이 있는지 파악하기 좋을 것입니다.

코드의 재사용성을 고려해도 하나의 빼빼로 만들기 함수로 모든 빼빼로를 만들 수 있어야하니까요!

또한 현재 순서도가 몸통을 그리는 부분까지밖에 나와있지 않습니다! 손잡이를 그리는 흐름까지 순서도에 나와야 할 것 같습니다.

Commit:
Week1 [STEP 3] Kobe

 @Gundy93
 건디 :) 너무 친절한 피드백 덕분에 이번에는 정말 빠르게 수정해서 PR 드립니다.

 ## 고민했던 점
 ### 반환 함수의 마지막 return 값
 - String으로 반환하는 함수가 있는데 마지막 글로벌 return 값을 "ERROR"로 처리했습니다.
   이게 맞는건지 궁굼합니다.

 ---

 ## 해결이 되지 않은 점
 ### 위 고민했던 점과 동일합니다.
 - 마지막 리턴 값을 어떻게 처리해야 할 지 모르겠습니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 조언 보다는 건디는 어떤 취미를 가지고 계신지 궁굼합니다 ㅎㅎ
 - 저는 배스 낚시를 좋아합니다. 데이터와 확률의 게임이라서 너무 재미있습니다.
   또 저는 보드게임도 좋아합니다. 건디는 어떤 취미를 가지고 계신가요??

Gundy93 reviewed (commented):
코비님 기능은 거의 수정된 것 같습니다!

> 반환 함수의 마지막 return 값

해당 고민은 `topping`이 옵셔널이기때문에 그것을 옵셔널 바인딩하는 과정에서 생기는 문제인 것 같습니다.

닐 병합 연산자를 사용해보면 어떨까요?

> 취미

요즘엔 잘 즐기지 못하지만 프라모델 만드는 취미가 있습니다

간단한 수정만 하고 다음 스텝으로 넘어갑시다!

Commit:
Week1 [STEP 3] Kobe

 @Gundy93
 건디 수정 후 PR 드립니다.
 건디의 피드백 덕분에 코드가 한층 더 깔끔해졌습니다 :)
 항상 감사합니다~

 ## 고민했던 점
 ### 코드의 가독성
 - 어떻게하면 조금 더 내가 아닌 다른 사람이 내 코드를 읽고 이해하기 쉽게 만들 수 있을까? 하고 고민해봤습니다.

 ---

 ## 해결이 되지 않은 점
 ### 위 고민과 동일합니다.
 - 아직은 코딩 스타일이 정립되지도 않았고 가독성 측면에서 나쁘다고 생각이 듭니다.
   코드를 가독성 있게 만들기 위해서 컨벤션을 공부하려고 합니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 코딩 컨벤션 부분
 - 코딩 컨벤션은 회사마다 혹은 프로젝트마다 다르다고 들었습니다.
   그래도 어떤 대표적으로 보고 공부할 수 있을 코딩 컨벤션 자료나 대표적인 코딩 컨벤션에 대한
   글이 있을까요??

 ### 코드 가독성 부분
 - 건디는 코드 가독성을 위해 어떤 노력을 하시는지 궁굼합니다.
